src/Providers/Stripe/Service.php

```diff
@@ -24,25 +29,40 @@ public function __construct(string $key)
      *
      * @param string $token
      * @param int $amount
+     * @param string|null $customerId
+     * @param string|null $description
+     * @param array $metadata
      * @param string $currency
      *
      * @return Payment
      *
-     * @throws Exception
+     * @throws PaymentException
      * @throws StripeException
      */
-    public function create(string $token, int $amount, string $currency = 'GBP'): Payment
-    {
+    public function create(
+        string $token,
+        int $amount,
+        ?string $customerId = null,
+        ?string $description = null,
+        array $metadata = [],
+        string $currency = 'GBP'
```

Comment:
not sure of the ordering here - I think maybe `currency` should still come after amount